evalml/pipelines/components/transformers/preprocessing/featuretools.py

```diff
@@ -42,6 +45,24 @@ def _make_entity_set(self, X):
             es = ft_es.add_dataframe(dataframe=X, dataframe_name="X", index=self.index)
         return es
 
+    def _filter_features(self, X):
+        features_to_use = []
+        for feature in self.features:
+            input_cols = [f.column_name for f in feature.base_features]
+            if (
```

Comment:
Should this be `and feature.columns_name in X.columns`? I think this is causing a bug where only the identify features are returned if you pass in data with identity and non-identity features?

```python
from evalml.demos import load_diabetes
import featuretools as ft
from evalml.pipelines.components import DFSTransformer
import pandas as pd
import pytest


X, y = load_diabetes()
del X.ww

X_fit = X.iloc[:X.shape[0]//2]
X_transform = X.iloc[X.shape[0]//2:]


es = ft.EntitySet()
es = es.add_dataframe(
    dataframe_name="X", dataframe=X_fit, index="index", make_index=True
)
feature_matrix, features = ft.dfs(
    entityset=es, target_dataframe_name="X", trans_primitives=["absolute"]
)
#features = list(filter(lambda f: not isinstance(f, IdentityFeature), features))

dfs = DFSTransformer(features=features)
dfs.fit(X_fit)
X_t = dfs.transform(feature_matrix)

with pytest.raises(AssertionError):
    pd.testing.assert_frame_equal(X_t, feature_matrix)
```